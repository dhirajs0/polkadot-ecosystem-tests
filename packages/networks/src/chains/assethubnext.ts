import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNext: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNext) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNext = defineChain({
  name: 'assetHubNext',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNext,
  initStorages: getInitStorages(custom.assetHubNext),
})
