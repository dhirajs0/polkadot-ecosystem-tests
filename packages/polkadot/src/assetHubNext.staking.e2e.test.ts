import { assetHubNext } from '@e2e-test/networks/chains'

import { stakingE2ETests } from '@e2e-test/shared'

stakingE2ETests(assetHubNext, { testSuiteName: 'AHN Staking', addressEncoding: 0 })
