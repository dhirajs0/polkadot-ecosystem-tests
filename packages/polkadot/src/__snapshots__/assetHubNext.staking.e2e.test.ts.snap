// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AHN Staking > chill other > chill other bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x01000000",
          "index": 80,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 2300000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`AHN Staking > force apply validator commission > force apply min commission events 1`] = `
[
  {
    "data": {
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 1400000000)",
        },
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;

exports[`AHN Staking > modify validator count > set validator count bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 1400000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`AHN Staking > set minimum validator commission > set staking configs bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 1400000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`AHN Staking > set staking configs > set staking configs bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 2000000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`AHN Staking > staking lifecycle > validator 0 bond events 1`] = `
[
  {
    "data": {
      "amount": 50000000000000,
      "stash": "5EtRedtmy3HcGPZU1BYRa4P7PiaNAeH9gME9vRDyr2S16A32",
    },
    "method": "Bonded",
    "section": "staking",
  },
]
`;

exports[`AHN Staking > staking lifecycle > validator 1 bond events 1`] = `
[
  {
    "data": {
      "amount": 50000000000000,
      "stash": "5GP5GRyw5gC2Z6DKcNkx1LkMaYSavJFGCicKZEcAfhvGNuxg",
    },
    "method": "Bonded",
    "section": "staking",
  },
]
`;

exports[`AHN Staking > staking lifecycle > validator 2 bond events 1`] = `
[
  {
    "data": {
      "amount": 50000000000000,
      "stash": "5ELp7TfqMPH65Auv2frN5UBA11mVKJEN7QUfUYNejDXeTtxF",
    },
    "method": "Bonded",
    "section": "staking",
  },
]
`;

exports[`AHN Staking > test fast unstake > nominate events 1`] = `[]`;

exports[`AHN Staking > test fast unstake > nominator bond events 1`] = `
[
  {
    "data": {
      "amount": 100000000000000,
      "stash": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
    },
    "method": "Bonded",
    "section": "staking",
  },
]
`;

exports[`AHN Staking > test force unstaking of nominator > force unstake bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 3000000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`AHN Staking > trying to become a validator with no bonded funds fails > events when attempting to validate with no bonded funds 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x00000000",
          "index": 80,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 2100000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`AHN Staking > trying to nominate with no bonded funds fails > events when attempting to nominate with no bonded funds 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x00000000",
          "index": 80,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 2300000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;
